Framework/Algorithms/src/PDFFourierTransform2.cpp

```diff
@@ -172,14 +172,19 @@ size_t PDFFourierTransform2::determineMinIndex(double min, const std::vector<dou
   }
 
   // get index for the min from the X-range
-  auto iter = std::upper_bound(X.begin(), X.end(), min);
+  auto iter = std::lower_bound(X.begin(), X.end(), min);
   size_t min_index = std::distance(X.begin(), iter);
-  if (min_index == 0)
-    min_index += 1; // so there doesn't have to be a check in integration loop
 
   // go to first non-nan value
-  iter = std::find_if(std::next(Y.begin(), min_index), Y.end(), static_cast<bool (*)(double)>(std::isnormal));
-  size_t first_normal_index = std::distance(Y.begin(), iter);
+  auto isNormalOrZero = [](double elem) {
```

Comment:
I think the original behaviour where it also skipped over zero y values was probably a useful feature - certainly for histogram data (ie bin edges). The integral for bins with y=0 is also zero so it made sense just to skip over them.

For point data, the integration uses a "left leaning" approach where it takes the y value from the left point for the integral between each pair of points (L498). Again it makes sense here to skip over points with y=0 but in the determineMaxIndex function the logic should perhaps find the farthest "right" non-zero point and then move one further on.